application/apps/indexer/sources/src/factory.rs

```diff
@@ -80,15 +98,30 @@ pub struct TCPTransportConfig {
     pub bind_addr: String,
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct UDPTransportConfig {
     pub bind_addr: String,
     pub multicast: Vec<MulticastInfo>,
 }
 
-///
 #[derive(Debug, Serialize, Deserialize)]
-pub enum SourceType {
-    File(PathBuf, ParserType),
-    Stream(Transport, ParserType),
+pub enum Sources {
```

Comment:
Can we call it s.th. like `ObserveOrigin` 
* less confusing with `ByteSource`
* enum name is singular